### ISSUE TITLE
[TRoW][Campaign][Units] Update Haldric Sprites

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#7121]
### Wesnoth 1.4 (March 2008)
<img width="688" alt="Screenshot 2022-11-13 at 8 55 42 PM" src="https://user-images.githubusercontent.com/5283677/201522734-da5aaa7e-96aa-4d84-9b10-c7dcbde866fc.png">

### Wesnoth 1.16 (October 2021)
<img width="706" alt="Screenshot 2022-11-13 at 8 56 31 PM" src="https://user-images.githubusercontent.com/5283677/201522769-555ce592-a93f-4991-bc9d-ba7726016a74.png">

### Summary:
As it can be seen above, the graphics for Haldric has not been updated for over...`2022-2008 =` 14 years. Thus, a graphics update would be nice to see.

**Comment by cooljeanius** [comment]
> ### Describe the desired feature
> ## Context
> ### Wesnoth 1.4 (March 2008)
> 
> <img alt="Screenshot 2022-11-13 at 8 55 42 PM" width="688" src="https://user-images.githubusercontent.com/5283677/201522734-da5aaa7e-96aa-4d84-9b10-c7dcbde866fc.png">
> ### Wesnoth 1.16 (October 2021)
> 
> <img alt="Screenshot 2022-11-13 at 8 56 31 PM" width="706" src="https://user-images.githubusercontent.com/5283677/201522769-555ce592-a93f-4991-bc9d-ba7726016a74.png">
> ### Summary:
> 
> As it can be seen above, the graphics for Haldric has not been updated for over...`2022-2008 =` 14 years. Thus, a graphics update would be nice to see.

Haldric is perfect as-is; he hasn't seen any updates in 14 years because he doesn't need any.

**Comment by Wedge009** [comment]
I agree that an update just for the sake of age isn't a reason in itself. However, there's a similar request in #6815 which I think is justified because the sprites that Deoran is based on have since been updated.

Haldric's portrait was last updated in 1.9.2 (6c523dee339136f9f0ccb34375310591173c0595). Perhaps the sprites could also be updated to match them better (similar to how Li'sar's portraits and then sprites were also updated, although Li'sar's change in appearance was a lot more dramatic than Haldric's change). I wouldn't call this a high priority, however.

**Comment by AnekronCZ** [comment]
I would say an update should be at least considered and I'll try to provide some reasons:

![image](https://github.com/wesnoth/wesnoth/assets/149104469/b6af91bc-4723-497b-9cec-a0286559df83)
- next to some current loyalist sprites, Haldric doesn't even look like lvl 3 unit anymore; he's far too small, barely larger than even a spearman
- it gets even more apparent when Haldric stands next to a mounted unit; for precisely that reason I rather swapped the current sprite in my UMC for Vyncyn's from the forums, even though that one doesn't have animations
- the sprite doesn't respect TC
- there's no animation for Haldric using the Ruby of Fire, while Konrad gets one for the Sceptre. If there was a new sprite, it could be made with that animation on top of the current ones
- some animations are of questionable quality, namely the defense sprites; Haldric just holds the sword next to his head for some reason

I agree with egallager that Haldric's look is iconic and I think any update should respect the original art and stick somewhat close to it (striped pants are a must!). Yet I feel there are good reasons to carefully improve on what we already have.

**Comment by cooljeanius** [comment]
>    * next to some current loyalist sprites, Haldric doesn't even look like lvl 3 unit anymore; he's far too small, barely larger than even a spearman

"Creeping biggerism" is a trend to be resisted, not accomodated.

>    * the sprite doesn't respect TC
> 
>    * there's no animation for Haldric using the Ruby of Fire, while Konrad gets one for the Sceptre. If there was a new sprite, it could be made with that animation on top of the current ones
> 
>    * some animations are of questionable quality, namely the defense sprites; Haldric just holds the sword next to his head for some reason
> 

These 3 points are all fair; I think an update that adds team coloring and some additional animation frames would be an improvement. 

> I agree with egallager that Haldric's look is iconic and I think any update should respect the original art and stick somewhat close to it (striped pants are a must!).

Besides the striped pants, I'd say that keeping his sword disproportionately oversized, in an anime-style fashion, is also important.

**Comment by MechanicalDragon963** [comment]
But his sword is quite proportionately big...

**Comment by Hejnewar** [comment]
He has no knees! That why he so short.

**Comment by cooljeanius** [comment]
> But his sword is quite proportionately big...

well, ok, proportionately or disproportionately, the point is, it should be big like Cloud Strife's Buster Sword
